pkg/cloudprovider/types.go

```diff
@@ -24,26 +24,15 @@ import (
 	"knative.dev/pkg/apis"
 )
 
-// Factory instantiates the cloud provider's resources
-type Factory interface {
-	// Capacity returns a provisioner for the provider to create instances
-	CapacityFor(provisioner *v1alpha1.Provisioner) Capacity
-	// Terminate allows the provider to terminate instances
-	TerminateFor() Termination
-}
-
-// Capacity provisions a set of nodes that fulfill a set of constraints.
-type Capacity interface {
+// API holds contains the methods necessary in a cloud provider
+type API interface {
 	// Create a set of nodes for each of the given constraints.
-	Create(context.Context, []*Packing) ([]*PackedNode, error)
+	Create(context.Context, []*Packing, *v1alpha1.Provisioner) ([]*PackedNode, error)
```

Comment:
Pass the provisioner first, imo.

Reply:
Before Packing, but not before context, right?